uniflow-core/src/main/kotlin/io/uniflow/core/flow/data/Event.kt

```diff
@@ -16,38 +16,35 @@
 
 package io.uniflow.core.flow.data
 
+import io.uniflow.core.flow.EventConsumer
+import io.uniflow.core.logger.UniFlowLogger
+import java.util.*
+
 /**
- * Data Flow Event wrapper
+ * Event wrapper
+ *
+ * Help send content to EventConsumer
  *
  * @author Arnaud Giuliani
  */
-data class Event<out T : UIEvent>(private val content: T) {
+data class Event<out T : UIEvent>(val id: String = UUID.randomUUID().toString(), val content: T) {
```

Comment:
What's the use for `val id: String`? It's not used anywhere, currently. It could be confusing for the user, because they might use it like so:

```kotlin
val uiEvent = object : UIEvent() {}

val a = Event(content = uiEvent)
val b = Event(content = uiEvent)

a == b // false!
```

Reply:
yes, let me check that 👍

Reply:
The user won't see the `Event` structure. It's under the hood. We ill just deal with `UIEvent`.

somehow it make sense to have ` a != b` as it's 2 different events, even if the data is the same. It would mean here that this is re-emitted.

Reply:
We could try to make it "internal"

Reply:
That would be nice, but I'm not sure if easily possible. As long as users don't receive `Event` instances, they won't need to do equality checks on them either.

Reply:
Yes, this is my idea also. To keep the `Event` wrapper class just as internal.